basic_module/distribution/src/impls.rs

```diff
@@ -0,0 +1,40 @@
+use crate::core::{Distribution, FeeManager};
+use crate::{Address, Deposit, StakeQuantity};
+use ckey::Ed25519Public as Public;
+use std::collections::{BTreeMap, HashSet};
+
+#[allow(dead_code)]
+struct DistributionState {
+    reward: IntermediateRewards,
+
+    // `validators` is used for inactive validators after distributing rewards
+    inactive_validators: HashSet<Address>,
+
+    // `next_validator` is used to store updated validator set to TopLevelState
```

Comment:
Why Distribution module manages validators in its state?
I think it should be managed by Staking module.